qutebrowser/browser/webkit/webpage.py

```diff
@@ -192,6 +192,17 @@ def _handle_errorpage(self, info, errpage):
             errpage.encoding = 'utf-8'
             return True
 
+    def chooseFile(self, parent_frame: QWebFrame, suggested_file: str) -> str:
+        """Override chooseFile to (optionally) invoke custom file uploader."""
+        if config.val.fileselect.handler == "default":
+            return super().chooseFile(parent_frame, suggested_file)
+
+        selected_file = shared.choose_file(multiple=False)
```

Comment:
nitpick: I see why you're calling this `selected_file`, but it sounds like this would be a single file rather than a list containing one file. Maybe just call it `selected`?

Reply:
Changed to `selected`